File: src/leetcode/word/LengthOfLastWord.java
package leetcode.word;

public class LengthOfLastWord {
    
    public int lengthOfLastWord(String s) {
        // IMPORTANT: Please reset any member data you declared, as
        // the same Solution instance will be reused for each test case.
        if(s == null || s.length() == 0)
            return 0;
        
        int len = s.length();
        int i = len - 1;
        while(i >= 0 && s.charAt(i) == ' ')
            i--;
        if(i == -1)
            return 0;
        
        int end = i;
        for(; i >= 0; i--){
            if(s.charAt(i) == ' ')
                break;
        }
        return end - i;
    }
}

File: cpractice/leetcode/length_of_last_word.cc
#include <string.h>

class Solution {
public:
    int lengthOfLastWord(const char *s) {
        int len = strlen(s);
        if(len == 0)    return 0;
        int res = 0;
        for(int i = len - 1; i >= 0; i--){
            if(s[i] != ' ')
                res++;
            else{
                if(res > 0)
                    return res;
            }
        }
        return res;
    }
};

int main(){
}